misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/completion/context/RemoteMethodCallActionNodeContext.java

```diff
@@ -0,0 +1,101 @@
+import org.eclipse.lsp4j.debug.CompletionItem;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.ballerinalang.debugadapter.completion.util.CommonUtil.getClientActions;
+import static org.ballerinalang.debugadapter.completion.util.CompletionUtil.getCompletionItemList;
+
+/**
+ * Handles the completions for the {@link RemoteMethodCallActionNode}.
+ *
+ * @since 2201.1.0
+ */
+public class RemoteMethodCallActionNodeContext {
+
+    /**
+     * Get Completion items for the scope/ context.
+     *
+     * @param context Debug completion context
+     * @param node    Node instance for the parser context
+     * @return {@link List}     List of calculated Completion Items
+     */
+    public List<CompletionItem> getCompletions(CompletionContext context, RemoteMethodCallActionNode node) {
+        List<CompletionItem> completionItems = new ArrayList<>();
+        ContextTypeResolver resolver = new ContextTypeResolver(context);
+        Optional<TypeSymbol> expressionType = node.expression().apply(resolver);
+
+        if (expressionType.isEmpty() || !SymbolUtil.isClient(expressionType.get())) {
+            return Collections.emptyList();
+        }
+
+        if (onSuggestClientActions(node, context)) {
+            /*
+            Covers the following case where a is a client object and we suggest the remote actions
+            (1) a -> g<cursor>
+             */
+            List<Symbol> clientActions = getClientActions(expressionType.get());
+            completionItems.addAll(getCompletionItemList(clientActions, context));
+        } else if (CommonUtil.isInMethodCallParameterContext(context, node)) {
+            /*
+             * Covers the following cases:
+             * 1. a->func(<cursor>)
+             * 2. a->func(mod1:<cursor>)
+             */
+            if (QNameReferenceUtil.onQualifiedNameIdentifier(context, node)) {
+                QualifiedNameReferenceNode qNameRef = (QualifiedNameReferenceNode) context.getNodeAtCursor();
+                List<Symbol> exprEntries = QNameReferenceUtil.getExpressionContextEntries(context, qNameRef);
+                List<CompletionItem> items = getCompletionItemList(exprEntries, context);
+                completionItems.addAll(items);
+            } else {
+                // TODO: Need to implement
+                // completionItems.addAll(this.actionKWCompletions(context));
+                // completionItems.addAll(this.expressionCompletions(context));
+                // completionItems.addAll(this.getNamedArgExpressionCompletionItems(context, node));
+            }
+        }
+
+        List<CompletionItem> completionItems1 = new ArrayList<>();
+
+        for (CompletionItem completionItem : completionItems) {
+            if (!completionItems1.contains(completionItem)) {
+                completionItems1.add(completionItem);
+            }
+        }
```

Comment:
Are we checking duplications here? We need to avoid such checks and handle them in a different manner. Need to avoid adding duplicate entries in the first place.